src/utils.ts

```diff
@@ -228,6 +242,16 @@ export namespace Utils {
           "The transaction has operations that are unrecognized",
         );
       }
+      if (
+        op.name === "web_auth_domain" &&
+        (!op.value || op.value.toString() !== webAuthDomain)
```

Comment:
❗ We're calling .toString() on a value we want to verify and comparing it with our expected value. I'm pretty sure the type of `op.value` is a `Buffer` and calling `toString()` without giving it an encoding will use `utf8`. The `toString()` function won't provide a unique output for every input because of this. It's not converting the bytes in the Buffer directly into a string that just holds those same bytes, it's interpreting the bytes into a string. It might replace invalid utf8 bytes with question marks, or it might return the same string for two technically different unicode sequences since unicode allows for representing the same string with different code points.

It's usually safer to not mutate the data to be verified, and instead get the expected value into the same format as the input data. The implementation is then less susceptible to the input data.